language-support/ts/daml-ledger/index.ts

```diff
@@ -710,8 +719,8 @@ class Ledger {
     // given key be in output format, whereas existing implementation supports
     // input format.
     let lastContractId: ContractId<T> | null = null;
-    const request = [{templateId: template.templateId, key}];
-    const reconnectRequest = (): object[] => [{...request[0], 'contractIdAtOffset': lastContractId}]
+    const request = [{templateId: template.templateId, key: template.keyEncode(key)}];
+    const reconnectRequest = (): object[] => [{...request[0], 'contractIdAtOffset': lastContractId && ContractId(template).encode(lastContractId)}]
```

Comment:
I feel like optional chaining should work for this somehow but I’m not sure how :thinking:

Reply:
It seems like [optional chaining in TypeScript](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html) is a bit object-centric: if what you want to do is access a property of the object (`o?.prop`), or call the object as a function (`o?.(args)`), it works, but I don't see a way to use it as an argument to something else. I may be missing something though; happy to take suggestions for making this better.

Reply:
Yeah I couldn’t figure out a way to use optional chaining here either. I don’t know why they don’t just have an `fmap` :disappointed: